src/workerd/server/server.c++

```diff
@@ -1304,79 +1304,72 @@ private:
   public:
-    CacheHttpClientImpl(Service& parent,
-                        kj::Maybe<kj::String> cacheName, kj::Maybe<kj::String> cfBlobJson,
-                        kj::Maybe<Tracer::Span&> parentSpan)
-        : cacheName(kj::mv(cacheName)),
-          cfBlobJson(kj::mv(cfBlobJson)), parentSpan(kj::mv(parentSpan)) {
-      client = asHttpClient(parent.startRequest({kj::mv(cfBlobJson), parentSpan}));
-    }
+    CacheHttpClientImpl(Service &parent, kj::Maybe<kj::String> cacheName,
+                        kj::Maybe<kj::String> cfBlobJson,
+                        kj::Maybe<Tracer::Span &> parentSpan)
+        : client(asHttpClient(
+              parent.startRequest({kj::mv(cfBlobJson), parentSpan}))),
+          cacheName(kj::mv(cacheName)) {}
 
     Request request(kj::HttpMethod method, kj::StringPtr url,
                     const kj::HttpHeaders &headers,
                     kj::Maybe<uint64_t> expectedBodySize = nullptr) override {
 
-      return client->request(method, url, addCacheNameHeader(headers, cacheName),
-                           expectedBodySize);
+      return client->request(method, url,
+                             addCacheNameHeader(headers, cacheName),
+                             expectedBodySize);
     }
 
   private:
     kj::Own<kj::HttpClient> client;
     kj::Maybe<kj::String> cacheName;
-    kj::Maybe<kj::String> cfBlobJson;
-    kj::Maybe<Tracer::Span&> parentSpan;
 
-    kj::HttpHeaders addCacheNameHeader(const kj::HttpHeaders& headers,
-       kj::Maybe<kj::StringPtr> cacheName) {
-      kj::ArrayPtr<const char> spliceParts[3];
-      uint spliceCount = 0;
+    kj::HttpHeaders addCacheNameHeader(const kj::HttpHeaders &headers,
+                                       kj::Maybe<kj::StringPtr> cacheName) {
       auto headersCopy = headers.cloneShallow();
-      KJ_IF_MAYBE(name, cacheName) {
-        spliceParts[spliceCount++] = "{\"namespace\":\""_kj;
-        spliceParts[spliceCount++] = *name;
-        spliceParts[spliceCount++] = "\"}"_kj;
-      } else {
-        spliceParts[spliceCount++] = "{}"_kj;
+      KJ_IF_MAYBE (name, cacheName) {
+        auto encodedName = kj::encodeUriComponent(*name);
+
+        headersCopy.set(IoContext::current().getHeaderIds().cfCacheNamespace,
+                        kj::mv(encodedName));
       }
-      headersCopy.set(IoContext::current().getHeaderIds().cfCacheMetadata, kj::strArray(kj::arrayPtr(spliceParts, spliceCount), ""));
+
       return headersCopy;
     }
   };
 
-
   kj::Own<CacheClient> getCache() override {
-     auto& channels = KJ_REQUIRE_NONNULL(ioChannels.tryGet<LinkedIoChannels>(),
-        "link() has not been called");
-    return kj::heap<CacheClientImpl>(*channels.cache);
+    auto &channels = KJ_REQUIRE_NONNULL(ioChannels.tryGet<LinkedIoChannels>(),
+                                        "link() has not been called");
+    auto &cache =
+        JSG_REQUIRE_NONNULL(channels.cache, Error, "No Cache was configured");
```

Comment:
`channels.cache` is not a `kj::Maybe`, it's a raw pointer. The "assert nonnull" asserts work on pointers but it's good style to always use a `Maybe` when nullness is possible.